builtin/mainmenu/dlg_settings_advanced.lua

```diff
@@ -658,11 +658,15 @@ local function create_change_setting_formspec(dialogdata)
 		height = height + 1.1
 
 	elseif setting.type == "flags" then
-		local enabled_flags = flags_to_table(get_current_value(setting))
+		local current_flags = flags_to_table(get_current_value(setting))
 		local flags = {}
-		for _, name in ipairs(enabled_flags) do
+		for _, name in ipairs(current_flags) do
 			-- Index by name, to avoid iterating over all enabled_flags for every possible flag.
-			flags[name] = true
+			if name.sub(1, 2) == "no" then
+				flags[name.sub(3)] = false
```

Comment:
Both times: Must be `name:sub` to pass the string as argument. This worked for you because `string.sub(1, 2)` returns `nil` in all cases.